core/src/air/extension.rs

```diff
@@ -0,0 +1,65 @@
+use std::ops::{Add, Mul, Neg, Sub};
+
+use p3_field::AbstractField;
+use sp1_derive::AlignedBorrow;
+
+const DEGREE: usize = 4;
+
+#[derive(AlignedBorrow, Clone, Copy, Debug, Default, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub struct Extension<T>(pub [T; DEGREE]);
```

Comment:
I don't think we have to use a struct for the extension, we could instead use the build in trait in Plonky3 and just case from an array.